tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -2868,28 +2920,15 @@ write_memory_order_and_alignment(INSTR_LIST *instrs)
 }
 
 INLINE static void
-write_no_depcheck_metadata(LL_Module *module, INSTR_LIST *insn)
-{
-  if (insn->flags & LDST_HAS_METADATA) {
-    char buf[64];
-    int n;
-    DEBUG_ASSERT(insn->misc_metadata, "missing metadata");
-    n = snprintf(buf, 64, ", !llvm.mem.parallel_loop_access !%u",
-                 LL_MDREF_value(insn->misc_metadata));
-    DEBUG_ASSERT(n < 64, "buffer overrun");
-    print_token(buf);
-  }
-}
-
-INLINE static void
-write_llaccgroup_metadata(LL_Module *module, INSTR_LIST *insn)
+write_llaccgroup_metadata(LL_Module *module, INSTR_LIST *insn,
+                          LL_InstrListFlags flag_check, unsigned int val)
 {
-  if (insn->flags & LDST_HAS_ACCESSGRP_METADATA) {
+  if (insn->flags & flag_check) {
```

Comment:
insn->flags seems to be not set when flag_check is LDST_HAS_ACCESSGRP_METADATA.

This ultimately happens because _mark_rw_access_grp_ is only called when 191, 0x4 is set (only if the directive is _vector always_). Could you call _mark_rw_access_grp_ for all of the cases where the access group metadata has to be generated??

```
    if (XBIT(191, 0x4)) {
      fix_nodepchk_flag(bih);
      mark_rw_access_grp(bih);
    }
```

Reply:
This has been addressed by adding the flag in the appropraite section.